asm/instruction.go

```diff
@@ -309,6 +309,26 @@ func (ins Instruction) Size() uint64 {
 // Instructions is an eBPF program.
 type Instructions []Instruction
 
+// Unmarshal unmarshals an Instructions from a binary instruction stream.
+func (insns *Instructions) Unmarshal(r io.Reader, bo binary.ByteOrder) error {
```

Comment:
Seems like calling this on non-empty `Instructions` will append to the instruction stream. Should we replace the instructions instead? (Maybe there is precedent for this behavior somewhere in the library?) If we append it should be mentioned in the doc I think.

Reply:
That's not intended, the expected behaviour attached to the term 'unmarshaling' is definitely to replace.